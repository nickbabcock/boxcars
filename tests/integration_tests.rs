use boxcars::ParserBuilder;
use highway::HighwayHash;
use insta::{assert_json_snapshot, glob};
use std::fs;

#[test]
fn test_replay_snapshots() {
    glob!("../assets/replays/good", "*.replay", |path| {
        let data = fs::read(path).unwrap();
        let replay = ParserBuilder::new(&data[..])
            .always_check_crc()
            .must_parse_network_data()
            .parse()
            .unwrap();

        // Hash the output otherwise we'll have 2.5GB+ of snapshot data
        let mut hasher = highway::HighwayHasher::new(highway::Key::default());
        serde_json::to_writer(&mut hasher, &replay).unwrap();
        let hash = hasher.finalize256();
        let out = hash
            .iter()
            .map(|x| format!("{:016x}", x))
            .collect::<String>();
        let hex = format!("0x{}", out);

        let snapshot = serde_json::json!({
            "frames": replay.network_frames.unwrap().frames.len(),
            "hex": hex
        });
        assert_json_snapshot!(snapshot);
    });
}
