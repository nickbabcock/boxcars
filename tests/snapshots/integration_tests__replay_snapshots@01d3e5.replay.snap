---
source: tests/integration_tests.rs
expression: snapshot
input_file: assets/replays/good/01d3e5.replay
---
{
  "frames": 393,
  "hex": "0xb19e81006218b16d38a75520fe9e46afc876e7d74a65b7131a29f6985ce06435"
}
